src/mozanalysis/metrics.py

```diff
@@ -141,16 +146,22 @@ def build_query(
         experiment_slug: str,
         from_expr_dataset: str | None = None,
         analysis_basis: str = AnalysisBasis.ENROLLMENTS,
+        experimental_unit: ExperimentalUnit = ExperimentalUnit.CLIENT,
         exposure_signal=None,
     ) -> str:
         """Return a nearly-self contained SQL query.
 
         This query does not define ``enrollments`` but otherwise could
         be executed to query all metrics from this data source.
         """
-        return """
-        SELECT
-            e.client_id,
+        if experimental_unit == ExperimentalUnit.CLIENT:
+            ds_id = self.client_id_column or "client_id"
+        elif experimental_unit == ExperimentalUnit.GROUP:
+            ds_id = self.group_id_column or "profile_group_id"
```

Comment:
It shouldn't be necessary to specify `or "client_id"`, right? The default is set on the class attr, so unless the config specifically defines it as an invalid falsy value, I don't know what this would do. I guess probably safer to just leave it in since this was the construction already... What do you think?

Reply:
I'm inclined to agree. As you said, I did it this way b/c that's how it already was done, but I don't _think_ `client_id_column` should ever be null...

Reply:
Actually, the `type=str` attrs parameter doesn't _validate_ the type at runtime. See below example for a metric pulled from `ConfigLoader` which has a `None` where we would expect a string
![Screenshot 2024-08-02 at 8 56 37 AM](https://github.com/user-attachments/assets/178a35ac-98ca-451f-a844-45bda20c6b72)

Reply:
I propose we move these `or default_value` guards to the attrs field definition, something like: 

```python
client_id_column = attr.ib(
    default=ExperimentalUnit.CLIENT.value,
    type=str,
    validator=[attr.validators.instance_of(str), attr.validators.min_len(1)],
    converter=lambda x: "client_id" if x is None else x
)
```

that way, anywhere inside the class, `self.client_id_column` is guaranteed to be a valid string

Reply:
ok, I made this change. the `or "client_id"` check can be removed, now guaranteed to be a string at runtime. Also made the change for group_id_column and submission_date_column

Reply:
Awesome, thanks! And I agree on the metric-config-parser suggestion as well.

Reply:
Great, [issue filed](https://github.com/mozilla/metric-hub/issues/545)